framework/src/base/ScalarCoupleable.C

```diff
@@ -84,6 +84,15 @@ ScalarCoupleable::coupledScalar(const std::string & var_name, unsigned int comp)
   return getScalarVar(var_name, comp)->number();
 }
 
+Order
+ScalarCoupleable::coupledScalarOrder(const std::string & var_name, unsigned int comp)
+{
+  if (!isCoupledScalar(var_name, comp))
+    return FIRST;
```

Comment:
If I set an optionally coupled variable in `validParams` and I do not couple this var in the input file and call `coupledScalarOrder(that_var)` in c-tor initializer, I am gonna get FIRST back, which is IMO wrong, there should be an error message.